pkg/controller/master/backup/restore.go

```diff
@@ -481,11 +480,6 @@ func (h *RestoreHandler) reconcileResources(
 		return nil, false, err
 	}
 
-	// mount volumes after creating PVCs, so detaching volumes controller doesn't detach the volumes
-	if err := h.mountLonghornVolumes(backup); err != nil {
-		return nil, false, err
-	}
-
```

Comment:
If my understanding is correct, since the restored pvc is from a backup volume's snapshot, hence during the new pvc creation we have to make sure the backup volume is attached.

[From the LH A/D implementation](https://github.com/longhorn/longhorn/blob/master/enhancements/20221024-longhorn-volumeattachment.md), we can make sure the volume would be attached during snapshot creation. However, in this case, can we make sure the volume would keep attached through new pvc provision?

Reply:
I think creating a PVC from a CSI Snapshot is also a CSI request. In `csi-provisioner` logs, I can see new PVC name. Following are my test steps:

1. In Harvester Volume page, create a volume `vol1`.
2. Take a snapshot for `vol1`, `vol1` state changes from `detached` to `attached`.
3. After the snapshot is finished, `vol1` state changes back to `detached`.
4. Provisioning a new PVC from the snapshot, `vol1` state changes from `detached` to `attached` again.
5. After a new PVC is provisioned, `vol1` state changes back to `detached`.

Reply:
That makes sense, thanks.